### ISSUE TITLE
[glsl-in] Infinite loop when first token of a statement is invalid.

### ISSUE THREAD

**Issue by fsh** [gfx-rs/naga#1277]
Something like the following hits an infinite loop in `parse_compound_statement()` because `parse_statement()` silently ignores invalid start tokens.
```
void main () {
 /
}
```

I fixed this locally by replacing the default case with
```
          _ => {
            let Token { value, meta } = self.bump(parser)?;
            return Err(Error {
              kind: ErrorKind::InvalidToken(
                value,
                vec![
                  TokenValue::RightBrace.into(),
                  // TODO: add all possible statement start tokens here? (has to be a better way)
                ],
              ),
              meta,
            });
          }
 ```

But since I'm not familiar with the rest of the parser I don't know if it's correct, i.e. are there valid cases where parse_statement() has to be a no-op or act like a silent peek? Any comment on that, or should I submit a PR?

**Comment by JCapucho** [comment]
If you can submit a PR it would be helpful :)

**Comment by JCapucho** [comment]
This was fixed in #1280
